bundles/org.openhab.binding.ventaair/src/main/java/org/openhab/binding/ventaair/internal/discovery/VentaDeviceDiscovery.java

```diff
@@ -0,0 +1,176 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ventaair.internal.discovery;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ventaair.internal.VentaAirBindingConstants;
+import org.openhab.binding.ventaair.internal.message.Header;
+import org.openhab.binding.ventaair.internal.message.Message;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.util.HexUtils;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Discovers Venta Air humidifier and cleaner devices by listening for UDP messages
+ *
+ * @author Stefan Triller - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery.ventaair")
+public class VentaDeviceDiscovery extends AbstractDiscoveryService {
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
+            .singleton(VentaAirBindingConstants.THING_TYPE_LW60T);
+    // defined as int, because AbstractDiscoveryService wants and int and not long as provided by Duration.getSeconds()
+    private static final int MANUAL_DISCOVERY_TIME = 30;
+
+    private final Logger logger = LoggerFactory.getLogger(VentaDeviceDiscovery.class);
+
+    private final Duration TIME_BETWEEN_SCANS = Duration.ofSeconds(30);
+
+    private @Nullable ScheduledFuture<?> scanJob = null;
+
+    public VentaDeviceDiscovery() {
+        super(SUPPORTED_THING_TYPES_UIDS, MANUAL_DISCOVERY_TIME, true);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("startScan() pressed");
+        findDevices();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        super.startBackgroundDiscovery();
+        logger.debug("startBackgroundDiscovery()");
+
+        ScheduledFuture<?> localScanJob = scanJob;
+        if (localScanJob != null && !localScanJob.isCancelled()) {
```

Comment:
No harm in canceling a task that is already cancelled.
```suggestion
        if (localScanJob != null) {
```